src/net/mod.rs

```diff
@@ -69,9 +69,15 @@ impl Connector {
 
     pub async fn connect(&self, addr: &str) -> Result<Stream> {
         match &self.inner {
-            ConnectorImpl::Tcp => Ok(Stream {
-                inner: StreamImpl::Tcp(tokio::net::TcpStream::connect(addr).await?),
-            }),
+            ConnectorImpl::Tcp => {
+                let s = tokio::net::TcpStream::connect(addr).await?;
+                match s.set_nodelay(true) {
+                    Ok(_) => Ok(Stream {
+                        inner: StreamImpl::Tcp(s),
+                    }),
+                    Err(x) => Err(x),
```

Comment:
If we do decide that erroring out is the best option, can change this to a `s.set_nodelay(true)?`

Reply:
Best-effort is probably fine, with debug logging before returning.

Reply:
Ack, that's probably the best option.